activerecord/lib/active_record/connection_adapters/mysql/schema_creation.rb

```diff
@@ -29,6 +29,9 @@ def column_options(o)
         end
 
         def add_column_options!(sql, options)
+          if /\Atimestamp\b/ === options[:column].sql_type && !options[:primary_key]
+            sql << " NULL" unless options[:null] == false || options_include_default?(options)
+          end
```

Comment:
We need to document this tricky behavior or nobody will understand it. MySQL auto-inserts current_timestamp on the first timestamp field in the table unless it's declared NULL.

http://dev.mysql.com/doc/refman/5.7/en/timestamp-initialization.html